### PR TITLE
draft for Article BFF

### PR DESCRIPTION
**Pull request by moritzzimmer**
PR to discuss the article (page) BFF used by paper (and future AMP) service to get relevant data.

- moved all paper related api to new folder `web`
- introduced separate `WebArticle` message
- moved core article grpc service next to proto message (needs a bit of work in polyphase)
- tried to used meaningful file names

### CODE REVIEW

File: stroeer/article/v1/service.proto [hunk 1]
```diff
@@ -10,9 +10,34 @@ option java_package = "de.stroeer.article.v1";
...
+  // rendering (canonical web or AMP) an Article with the given id.
+  rpc GetArticlePage (GetArticlePageRequest) returns (ArticlePage) {}
 }
 
 message GetArticleRequest {
   string id = 1;
 }
 
+message GetArticlePageRequest {
+  string id = 1;
```
Comment (thisismana) [inline stroeer/article/v1/service.proto:24]: what's the `id` in this case? A path like `/` or `/culture/`?

File: stroeer/article/v1/service.proto [hunk 2]
```diff
@@ -10,9 +10,34 @@ option java_package = "de.stroeer.article.v1";
...
+}
+
+message RelatedArticle {
+  // we could also use an stroeer.core.v1.Article here but maybe it's worth to have a Teaser abstraction
+  // including only teaser relevant attributes?!
+  stroeer.core.v1.Teaser teaser = 1;
+  RelatedArticleType role = 2;
+}
+
+enum RelatedArticleType {
```
Comment (thisismana) [inline stroeer/article/v1/service.proto:39]: `RelatedArticleRole` ?

File: stroeer/article/v1/service.proto [hunk 3]
```diff
@@ -10,9 +10,34 @@ option java_package = "de.stroeer.article.v1";
...
+  RELATED_ARTICLE_TYPE_UNSPECIFIED = 0;
```
Comment (thisismana) [inline stroeer/article/v1/service.proto:40]: Shouldn't we just filter out relations unknown by our code during transformation?

File: stroeer/article/v1/service.proto (hunk 1 above)
Comment (moritzzimmer) [reply stroeer/article/v1/service.proto]: the CMS id

File: stroeer/article/v1/service.proto (hunk 3 above)
Comment (moritzzimmer) [reply stroeer/article/v1/service.proto]: protobuf lint rule to always include unknown at `0`

File: stroeer/web/section/v1/web_section_service.proto [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import "stroeer/web/article/v1/web_article.proto";
+
+option java_multiple_files = true;
+option java_package = "de.stroeer.web.section.v1";
+
+service WebSectionService {
+  rpc GetSectionPage (GetSectionPageRequest) returns (GetSectionPageResponse) {}
+}
+
+message GetSectionPageRequest {
```
Comment (harryurban) [inline stroeer/web/section/v1/web_section_service.proto:14]: for now, I guess?
Will need to change, as soon as we have stages, with labels, and configs for the page.

**Pull request merged**